third-party/sysdeps/common/liblzma/CMakeLists.txt

```diff
@@ -0,0 +1,141 @@
+      EXTERNAL_SOURCE_DIR .
+      BINARY_DIR build
+      NO_MERGE_COMPILE_COMMANDS
+      BACKGROUND_BUILD
+      OUTPUT_ON_FAILURE
+      CMAKE_ARGS
+        "-DSOURCE_DIR=${_source_dir}"
+        "-DPATCHELF=${PATCHELF}"
+        "-DPython3_EXECUTABLE=${Python3_EXECUTABLE}"
+        "-DLIBLZMA_VERSION=${LIBLZMA_VERSION}"
+      INSTALL_DESTINATION
+        lib/rocm_sysdeps
+      INTERFACE_PROGRAM_DIRS
+        lib/rocm_sysdeps/bin
+      INTERFACE_LINK_DIRS
+        lib/rocm_sysdeps/lib
+      INTERFACE_INSTALL_RPATH_DIRS
+        lib/rocm_sysdeps/lib
+      INTERFACE_PKG_CONFIG_DIRS
+        lib/rocm_sysdeps/lib/pkgconfig
+      EXTRA_DEPENDS
+        "${_download_stamp}"
+    )
+    therock_cmake_subproject_provide_package(therock-liblzma liblzma lib/rocm_sysdeps/lib/cmake/liblzma)
+    therock_cmake_subproject_activate(therock-liblzma)
+
+    therock_test_validate_shared_lib(
+      PATH build/dist/lib/rocm_sysdeps/lib
+      LIB_NAMES liblzma.so
+    )
+    return()
+endif()
+
+# Otherwise, this is the sub-project build.
+cmake_minimum_required(VERSION 3.25)
+project(LIBLZMA_BUILD VERSION "${LIBLZMA_VERSION}")
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND NOT PATCHELF)
+  message(FATAL_ERROR "Missing PATCHELF from super-project")
+endif()
+
+# Apply patch source only on Linux
+set(patch_source_commands)
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+  list(APPEND patch_source_commands   COMMAND
+    bash "${CMAKE_CURRENT_SOURCE_DIR}/patch_source.sh" "${CMAKE_CURRENT_BINARY_DIR}/s")
+endif()
+# xz/liblzma provides a CMakeLists.txt, however, we have to do some post-processing
+# of the libraries in order to prepare them for our use, so we invoke it as
+# a sub-build. We do this uniformly across all platforms because it is easier
+# to have common code paths. We write directly into the install prefix at
+# build time since this build is only ever used in a one shot way.
+add_custom_target(
+  build ALL
+  WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
+  COMMAND
+    "${CMAKE_COMMAND}" -E rm -rf -- "${CMAKE_INSTALL_PREFIX}" "${CMAKE_CURRENT_BINARY_DIR}/s"
+  COMMAND
+    "${CMAKE_COMMAND}" -E copy_directory "${SOURCE_DIR}" "${CMAKE_CURRENT_BINARY_DIR}/s"
+  COMMAND
+    ${patch_source_commands}
+  COMMAND
+    # Provide our own version map with private symbol versions.
+    "${CMAKE_COMMAND}" -E copy "${CMAKE_CURRENT_SOURCE_DIR}/version.lds" "${CMAKE_CURRENT_BINARY_DIR}/s/src/liblzma/liblzma_generic.map"
+  COMMAND
+    "${CMAKE_COMMAND}"
+      "-G${CMAKE_GENERATOR}"
+      "-S${CMAKE_CURRENT_BINARY_DIR}/s"
+      "-B${CMAKE_CURRENT_BINARY_DIR}/b"
+      "-DCMAKE_PROJECT_TOP_LEVEL_INCLUDES=${CMAKE_CURRENT_SOURCE_DIR}/liblzma_include.cmake"
+      "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
+      "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
+      "-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}"
+      "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}"
+      "-DCMAKE_BUILD_TYPE=Release"
+      # Force shared library build
+      "-DBUILD_SHARED_LIBS=ON"
```

Comment:
Pretty sure you only want to be building shared libs on Linux. Want static on Windows like the others, right?

Reply:
(looks below like you are managing this explicitly and can just omit this as it is not used?)

Reply:
thanks, i think that was some leftover of getting the symbol renaming working. 
i will remove it here and keep it in `third-party/sysdeps/common/liblzma/CMakeLists.txt`